css/backend/src/main/scala/io/udash/css/CssFileRenderer.scala

```diff
@@ -0,0 +1,40 @@
+package io.udash.css
+
+import java.io.{File, PrintWriter}
+
+import scalacss.internal.Renderer
+
+/**
+  * Renders provided styles into files. Creates separate file for each stylesheet.
+  * Keeps styles order from provided `Seq`.
+  * @param dirPath Target directory for rendered files.
+  * @param styles Sequence of stylesheets.
+  * @param createMain If true, creates `main.css` file. It imports all other stylesheets.
+  */
+class CssFileRenderer(dirPath: String, styles: Seq[_ <: CssBase], createMain: Boolean) {
+  def render()(implicit renderer: Renderer[String]): Unit = {
+    val dir = new File(dirPath)
+    dir.mkdirs()
+
+    val mainFile: Option[File] = if (createMain) Some(new File(s"${dir.getAbsolutePath}/main.css")) else None
+    mainFile.foreach(_.createNewFile())
+
+    val mainWriter = mainFile.map(f => new PrintWriter(f, "UTF-8"))
+
+    styles.foreach { style =>
+      val name = style.getClass.getName
+      val f = new File(s"${dir.getAbsolutePath}/$name.css")
+      f.createNewFile()
+      new PrintWriter(f, "UTF-8") {
```

Comment:
I did not ever think of using this syntax for scoping calls to an instance in an anonymous constructor, nice.
Maybe you should do the same with the `new File`s in this class?